src/GitVersion.Core/VersionCalculation/SemanticVersioning/SemanticVersionPreReleaseTag.cs

```diff
@@ -84,10 +84,10 @@ public override int GetHashCode()
 
         public static implicit operator SemanticVersionPreReleaseTag(string preReleaseTag)
         {
-            return Parse(preReleaseTag);
+            return TryParse(preReleaseTag);
         }
 
-        public static SemanticVersionPreReleaseTag Parse(string preReleaseTag)
+        public static SemanticVersionPreReleaseTag TryParse(string preReleaseTag)
```

Comment:
Since this method does not have the `bool TryParse(string, out T)` signature, I'm not sure it should be named `TryParse`. The method signature indicates `Parse` while its behaviour indicates `TryParse`. How much breakage would it cause to change the method signature to `bool TryParse(string, out SemanticVersionPreReleaseTag)`?

```suggestion
        public static bool TryParse(string preReleaseTag, out SemanticVersionPreReleaseTag)
```

Reply:
The `Int.TryParse()` in the `SemanticVersionPreReleaseTag` defaults the value to null if it's not parseable.

Will let you be the guide here as to whether you would prefer to rename the method it back to Parse, return false on failure and modify the signature as you have suggested above or take the same approach as in `SemanticVersion.cs`

In terms of breakage the only place it seems to be directly called is within `SemanticVersion.cs` when it constructs and returns a new `SemanticVersion` and assigns a pre-release tag. 

The tests do not seem to directly reference this method directly either.

From there there we can decide if we want to do anything about it eg. returning false from `SemanticVersion.TryParse()` which would result in the calling method `Parse()` throw a new Warning exception.

NB. All of the code calls `SemanticVersion.Parse()` nothing seems to call the `TryParse()` directly. I could follow the same pattern in `SemanticVersionPreReleaseTag` and wrap the method?

Calling Code in `TryParse()` 

```c#
            semanticVersion = new SemanticVersion
            {
                Major = major,
                Minor = minor,
                Patch = patch,
                PreReleaseTag = SemanticVersionPreReleaseTag.TryParse(parsed.Groups["Tag"].Value),
                BuildMetaData = semanticVersionBuildMetaData
            };`

Reply:
I would prefer consistent use of `TryParse` returning `bool` everywhere, if possible.